libs/shared/rank/src/lib/shared-rank.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+import { Box } from '@material-ui/core';
+import { Rating } from '@material-ui/lab';
+
+import './shared-rank.scss';
+
+export interface SharedRankProps {
+  value: number;
+  isRanked: boolean;
+  onUpdate: (data) => void;
+}
+
+export const SharedRank = (props: SharedRankProps) => {
+  const [rank, setRank] = useState({
+    isRanked: false,
+    value: null,
+  });
+
+  useEffect(() => {
+    setRank({ ...props });
+  }, [props]);
+
+  return (
+    <div className="rank">
+      <Rating
+        disabled={rank.isRanked}
+        name="hover-feedback"
+        value={rank.value}
+        precision={0.5}
+        onChange={(event: EventTarget, newValue: number) => {
+          if (!rank.isRanked) {
+            let newRank = { isRanked: true, value: newValue };
```

Comment:
I would recommend to write in `const` first and if the var is going to change, change to `let` 
![image](https://user-images.githubusercontent.com/36937039/87696604-f3eb8680-c788-11ea-9e2e-afc29a775017.png)